style: Refactor edtlib module code style

rruuaanng opened the pull request:
Move the content of the if statement block to
the next line, rather than keeping it on the
same line as the if statement.

pdgendt commented:
Looks good, might be worth to combine some of these fixes into 1 PR.

rruuaanng commented:
Yes, So, next PR, I will put the changes together. Repeated opening of PR makes me a little tired.

pdgendt commented:
> Yes, So, next PR, I will put the changes together. Repeated opening of PR makes me a little tired.

Why not append commits to this PR and update the title?

rruuaanng commented:
It seems that we cannot migrate some `Union` types to the `|` operator, which is causing failures in the current CI due to different versions of the interpreter.

pdgendt commented:
> It seems that we cannot migrate some `Union` types to the `|` operator, which is causing failures in the current CI due to different versions of the interpreter.

We need to make sure the minimum version (3.10) is still supported.

rruuaanng committed:
style: edtlib: Remove unnecessary quotes

Remove unnecessary quotes from deferred type
annotations to fix the Ruff UP037 warning.

Signed-off-by: James Roy <rruuaanng@outlook.com>

rruuaanng committed:
style: edtlib: Remove deprecated packages

Remove imported deprecated packages and replace
with new packages to fix Ruff UP035 warning.

Signed-off-by: James Roy <rruuaanng@outlook.com>

rruuaanng committed:
style: edtlib: Add an explicit strict argument to the zip

Add an explicit strict parameter to the zip
function to fix the Ruff B905 warning.

Signed-off-by: James Roy <rruuaanng@outlook.com>

rruuaanng committed:
style: edtlib: Sort the import statements

Sort the import statements to fix the
Ruff I001 warning.

Signed-off-by: James Roy <rruuaanng@outlook.com>

rruuaanng committed:
style: edtlib: Use a more readable not in statement

Replace 'not xx in xx' with the more readable
'xx not in xx' syntax to fix the Ruff E713 warning.

Signed-off-by: James Roy <rruuaanng@outlook.com>

rruuaanng committed:
style: edtlib: Merge redundant if statements

Merge redundant if statements to fix the Ruff
SIM102 warning.

Signed-off-by: James Roy <rruuaanng@outlook.com>

rruuaanng committed:
style: edtlib: Remove the redundant call arguments

Remove the redundant positional argument in
open function to fix the Ruff UP015 warning.

Signed-off-by: James Roy <rruuaanng@outlook.com>

rruuaanng committed:
style: edtlib: Reorganize the single-line if statement block

Move the statement in the single-line if block
to a new line to fix the Ruff E701 warning.

Signed-off-by: James Roy <rruuaanng@outlook.com>

rruuaanng committed:
style: edtlib: Use the more readable not-equal operator

Replace 'not xx == xx' with 'xx != xx' to
fix the Ruff SIM201 warning.

Signed-off-by: James Roy <rruuaanng@outlook.com>

rruuaanng committed:
style: edtlib: Use from to chain the exception

Use the from keyword to chain exceptions in
order to fix the Ruff B904 warning.

Signed-off-by: James Roy <rruuaanng@outlook.com>

rruuaanng committed:
style: edtlib: Remove the redundant keys() method

Remove the redundant keys method when using the
in statement on a dict to fix the Ruff SIM118
warning.

Signed-off-by: James Roy <rruuaanng@outlook.com>

rruuaanng commented:
@kartben If you have time, you can take a look at this, I think it's ready. :)

mbolivar commented on scripts/dts/python-devicetree/src/devicetree/edtlib.py:
```diff
@@ -2113,7 +2113,7 @@ def scc_order(self) -> list[list[Node]]:
             return self._graph.scc_order()
         except Exception as e:
-            raise EDTError(e)
+            raise EDTError(e) from None
```
Shouldn't this be `from e`?

mbolivar reviewed (approved):
LGTM, one non-blocking question
